kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -71,11 +78,18 @@ public List<Path> getCorrespondingFiles(Path path) {
                 Collections.singletonList(FileNames.checkpointFileSingular(path, version)));
     }
 
+    /**
+     * Comparison rules:
+     * 1. A [[CheckpointInstance]] with higher version is greater than the one with lower version.
```

Comment:
`[[CheckpointInstance]]` is scala-style, not java. Just say `CheckpointInstance` or `{@link CheckpointInstance}`